Automattic/jetpack
# Update/react router 5

Fixes #15502 #11262

#### Changes proposed in this Pull Request:

* Switch to `react-router-dom` v5
* Get rid of unused redux integration
* Get rid of `history` dependency

#### Testing instructions:






* Browser history works as expected
* Settings search, both via search box, and direct access
* Confirmation dialog on redirects from unsaved settings page

This PR affects few aspects of Jetpack dashboard routing mechanism. So I'll split testing instructions into sections.
1. Browser history
- Make sure moving back and forth around the dashboard still works.
- Navigation history persists even if you leave the dashboard.

2. Settings search
- Search via search box still works. Expected results are being rendered.
- Direct navigation to URL with search terms also works as expected. I.E `wp-admin/admin.php?page=jetpack#/settings?term=backup`

3.  Confirmation dialog on redirects from the unsaved settings page. There are 2 types of navigation: Browser, and Router. The current confirmation dialog in Browser nav does not work properly in major browsers, so I haven't tried to make it work in this PR.
- Changing any setting that does not save automatically should trigger a confirmation dialog on navigating away from the page (but only around the dashboard, moving away from `/wp-admin/admin.php?page=jetpack` won't trigger the dialog). 
- Canceling the confirmation should stop the redirect, unsaved changes should be there
- Confirming redirect should clean up the `dirty` state, so another redirect should not trigger the confirmation. 

4. Also, it would be nice to test setup wizard flow by following testing instructions from https://github.com/Automattic/jetpack/pull/15429

#### Proposed changelog entry for your changes:

* Switch to `react-router-dom` v5
